File: c_plus/Plus_One.cpp
#include <iostream>
#include <vector>
using namespace  std;

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry = 1;
        vector<int> res;
        while (digits.size() > 0) {
            int x = digits.back();
            digits.pop_back();
            x += carry;
            res.insert(res.begin(), x % 10);
            carry = x / 10;
            
        }
        if (carry > 0) 
            res.insert(res.begin(), 1);
        return res;
    }
};

void printVector(vector<int> &v) {
    cout << "{";
    for (int i = 0; i < v.size(); i++) {
        cout << v[i];
        if (i != v.size() - 1)
            cout << ", ";
    }
    cout << "}" << endl;
}

int main () {
    Solution s;
    vector<int> l = {1, 2, 3};
    vector<int> v = s.plusOne(l);
    printVector(v);
    return 0;
}

File: py/Plus_One.py
from typing import List

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        ls = len(digits)
        for index in reversed(range(ls)):
            if digits[index] < 9:
                digits[index] += 1
                # do not need to continue
                return digits
            else:
                # 10
                digits[index] = 0
        digits.insert(0, 1)
        return digits

if __name__ == '__main__':
    s = Solution()
    print(s.plusOne([1,2,3]))